test/Microsoft.DocAsCode.Build.Engine.Tests/DocumentBuilderTest.cs

```diff
@@ -893,3 +949,50 @@ private void CleanUp()
         }
     }
 }
+
+namespace UnitTestUtilities
+{
+    using System.Reflection;
+    using System;
+
+    public class Helper
+    {
+        public static object RunStaticMethod(System.Type t, string strMethod, object[] aobjParams)
+        {
+            BindingFlags eFlags =
+             BindingFlags.Static | BindingFlags.Public |
+             BindingFlags.NonPublic;
+            return RunMethod(t, strMethod,
+             null, aobjParams, eFlags);
+        } //end of method
+
+        public static object RunInstanceMethod(System.Type t, string strMethod, object objInstance, object[] aobjParams)
+        {
+            BindingFlags eFlags = BindingFlags.Instance | BindingFlags.Public |
+             BindingFlags.NonPublic;
+            return RunMethod(t, strMethod,
+             objInstance, aobjParams, eFlags);
+        } //end of method
+
+        private static object RunMethod(System.Type t, string strMethod, object objInstance, object[] aobjParams, BindingFlags eFlags)
+        {
+            MethodInfo m;
+            try
+            {
+                m = t.GetMethod(strMethod, eFlags);
+                if (m == null)
+                {
+                    throw new ArgumentException("There is no method '" +
+                     strMethod + "' for type '" + t.ToString() + "'.");
+                }
+
+                object objRet = m.Invoke(objInstance, aobjParams);
+                return objRet;
+            }
+            catch
+            {
+                throw;
+            }
```

Comment:
it preserves the stack trace

---
In reply to: [128892920](https://github.com/dotnet/docfx/pull/1837#discussion_r128892920) [](ancestors = 128892920)